localstack/aws/protocol/op_router.py

```diff
@@ -13,6 +13,11 @@
 
 
 class GreedyPathConverter(PathConverter):
+    """
+    This converter makes sure that the path ``/mybucket//mykey`` can be matched to the pattern
+    ``<Bucket>/<path:Key>`` and will result in `Key` being `/mykey`.
+    """
+
     regex = ".*?"
```

Comment:
Just for the record (and maybe as an extension to the "haunting" section in the PR description):
When I was looking at the issue I was initially suspecting it to originate from the fact that the converters now define if they are `part_isolating` or not, and it is implicitly determined by checking if the regex contains a `/` (see https://github.com/pallets/werkzeug/blob/e0afdfb0d549b4fea34e0f91270d784efe7b0e1b/src/werkzeug/routing/converters.py#L19-L36).
Here this custom converter does not have a `part_isolating` field defined, and does not contain a slash in the regex (contrary to the superclass), which leads to `part_isolating=False`.
In fact, this is the correct configuration, because this greedy path converter isn't only greedy for leading slashes, but can also match across multiple path segments.

I'm actually not really a fan of the implicit configuration here (it feels like we were lucky), but I also understand that Werkzeug does this to increase backwards compatibility. Maybe we should state this here explicitly?

Reply:
great find. i added the explicit `part_isolating=False` and also a comment for future us :grin: